core/src/main/scala/org/apache/spark/storage/BlockManagerMaster.scala

```diff
@@ -223,11 +223,11 @@ class BlockManagerMaster(
   }
 
   /**
-   * Find out if the executor has cached blocks. This method does not consider broadcast blocks,
-   * since they are not reported the master.
+   * Find out if the executor has cached blocks which are only available via this executor.
+   * This method does not consider broadcast blocks, since they are not reported the master.
    */
-  def hasCachedBlocks(executorId: String): Boolean = {
-    driverEndpoint.askSync[Boolean](HasCachedBlocks(executorId))
+  def hasExclusiveCachedBlocks(executorId: String): Boolean = {
```

Comment:
"exclusive" is not the greatest name for this.  The problem is that you can have a replicated, MEMORY_ONLY block.  So it could be on two executors (so its not exclusive to any one executor), but not owned by any shuffle service.

that said, I'm not sure of a better name.  It should at least be documented that non-exclusive means owned by the external shuffle service.